Pn 8027 fe rifattorizzare il componente empty state

NicolaGiornetta opened the pull request:
## Short description
Refactored EmptyState component and its usage.

## List of changes proposed in this pull request
- Removed most of the props about empty message structure with children prop in EmptyState component. Now you can use Trans component to make linear and complex empty messages.
- Reworked all the components which use EmptyState
- Reworked translations
- Reworked tests

## How to test
There are no visual changes because it should be the same. However you can test the purpose of EmptyState
Where to test:
- Notification list PA, PF, PG. Add a filter which makes not result.
- Delegates and delegators PA, PF, PG. Remove delegates / delegators to make empty list.
- API Keys in PA. Remove all the API keys to make empty list.
- App Status PA, PF, PG. Make sure to not have any disservice.

AndreaCimini90 commented on packages/pn-personafisica-webapp/src/components/Deleghe/Delegates.tsx:
```diff
@@ -113,6 +113,19 @@ const Delegates = () => {
     dispatch(setDelegatesSorting(s));
   };
 
+  const LinkAddDelegate = ({ children }: { children?: ReactNode }) => (
```
Same as before

AndreaCimini90 commented on packages/pn-personafisica-webapp/src/components/Deleghe/MobileDelegates.tsx:
```diff
@@ -107,6 +107,19 @@ const MobileDelegates = () => {
     setShowCodeModal({ open: false, name: '', code: '' });
   };
 
+  const LinkAddDelegate = ({ children }: { children?: ReactNode }) => (
```
Same as before

AndreaCimini90 commented on packages/pn-personafisica-webapp/src/components/Notifications/DesktopNotifications.tsx:
```diff
@@ -134,35 +135,36 @@ const DesktopNotifications = ({
-
   const showFilters = notifications?.length > 0 || filtersApplied;
 
+  const LinkRemoveFilters = ({ children }: { children?: ReactNode }) => (
```
Same as before

AndreaCimini90 commented on packages/pn-personagiuridica-webapp/src/components/Deleghe/DelegatesByCompany.tsx:
```diff
@@ -143,6 +144,19 @@ const DelegatesByCompany = () => {
     }
   };
 
+  const LinkAddDelegate = ({ children }: { children?: ReactNode }) => (
```
Same as before

AndreaCimini90 commented on packages/pn-personagiuridica-webapp/src/components/Deleghe/DelegationsOfTheCompany.tsx:
```diff
@@ -341,6 +342,19 @@ const DelegationsOfTheCompany = () => {
     getDelegatorsData();
   }, [filters]);
 
+  const LinkRemoveFilters = ({ children }: { children?: ReactNode }) => (
```
Same as before

AndreaCimini90 commented on packages/pn-personafisica-webapp/src/components/Notifications/DesktopNotifications.tsx:
```diff
@@ -192,7 +194,30 @@ const DesktopNotifications = ({
+          sentimentIcon={filtersApplied ? KnownSentiment.DISSATISFIED : KnownSentiment.NONE}
+        >
+          {filtersApplied && (
+            <Trans
```
I'm not sure that all the previous condition in EmptyStateProps are now respected

AndreaCimini90 commented on packages/pn-personafisica-webapp/src/components/Notifications/MobileNotifications.tsx:
```diff
@@ -254,7 +255,30 @@ const MobileNotifications = ({ notifications, sort, onChangeSorting, currentDele
+          sentimentIcon={filtersApplied ? KnownSentiment.DISSATISFIED : KnownSentiment.NONE}
+        >
+          {filtersApplied && (
+            <Trans
```
Same as before

AndreaCimini90 reviewed (changes requested):
Check also these things:
- In the page tests, where there is the rendering of the empty state, add the check on the action click
- With the rework of the EmptyState, maybe some localizations are been cancelled. Check this but for now DON'T remove them

NicolaGiornetta commented:
Translation labels replacements and additions:

notifiche.json of PA

removed:
"empty-state.message"
"empty-state.secondary-message"
"empty-state.secondary-action"
"empty-state.filter-message"
"empty-state.filter-action"

replaced with:
"empty-state.filtered" where before it was the sum of 
	"empty-state.message" +
	"menu.add" (of common.json) +
	"empty-state.secondary-message" +
	"empty-state.secondary-action"

"empty-state.no-notifications" where before it was the sum of
	"empty-state.filter-message" +
	"empty-state.filter-action"

"aria-label-remove-filters"
"aria-label-api-keys"
"aria-label-create-notification"

---------------------------------------------------------------------------

notifiche.json of PF

removed:
"empty-state.first-message"
"empty-state.action"
"empty-state.second-message"
"empty-state.filter-message"
"empty-state.filter-action"

replaced with:
"empty-state.filtered" where before it was the sum of
	"empty-state.filter-message" +
	"empty-state.filter-action"

"empty-state.no-notifications" where before it was the sum of
	"empty-state.first-message" +
	"empty-state.action" +
	"empty-state.second-message"
	
"empty-state.aria-label-remove-filters"
"empty-state.aria-label-route-contacts"

---------------------------------------------------------------------------

notifiche.json of PG

removed:
"empty-state.message"
"empty-state.filter-message",
"empty-state.filter-action"

replaced with:
"empty-state.filtered" where before it was the sum of
	"empty-state.filter-message" +
	"empty-state.filter-action"
	
"empty-state.no-notifications" where before it was "empty-state.message"
